### PR TITLE
[R4R] cdp rest fixes

### PR DESCRIPTION
**Pull request by karzak**
Minor fixes/cleanup to the cdp REST API. All tx and query endpoints tested on internal testnet.

### CODE REVIEW

File: x/cdp/client/rest/query.go [hunk 1]
```diff
@@ -16,11 +16,11 @@ import (
 
 // define routes that get registered by the main application
 func registerQueryRoutes(cliCtx context.CLIContext, r *mux.Router) {
-	r.HandleFunc(fmt.Sprintf("/cdp/{%s}/{%s}", types.RestOwner, types.RestCollateralDenom), queryCdpHandlerFn(cliCtx)).Methods("GET")
-	r.HandleFunc(fmt.Sprintf("/cdp/{%s}", types.RestCollateralDenom), queryCdpsHandlerFn(cliCtx)).Methods("GET")
-	r.HandleFunc(fmt.Sprintf("/cdp/{%s}/ratio/{%s}", types.RestCollateralDenom, types.RestRatio), queryCdpsByRatioHandlerFn(cliCtx)).Methods("GET")
-	r.HandleFunc(fmt.Sprintf("/cdp/deposits/{%s}/{%s}", types.RestOwner, types.RestCollateralDenom), queryCdpDepositsHandlerFn(cliCtx)).Methods("GET")
 	r.HandleFunc("/cdp/params", getParamsHandlerFn(cliCtx)).Methods("GET")
+	r.HandleFunc(fmt.Sprintf("/cdps/cdp/{%s}/{%s}", types.RestOwner, types.RestCollateralDenom), queryCdpHandlerFn(cliCtx)).Methods("GET")
```
Comment (denalimarsh) [inline x/cdp/client/rest/query.go]: Root should be `cdp` as opposed to `cdps` in order to match module name.

**Comment by denalimarsh** [comment]
Getting a weird error occasionally on endpoint '/cdps/cdp/{owner}/{denom}': `'{"codespace":"sdk","code":1,"message":"failed to load state at height 16765; version does not exist (latest height: 16881)"}' } }` which persists until I restart the rest server. Rest server works fine for a while until it snags on an incorrect block height with this query and has to be restarted again.

**Comment by denalimarsh** [comment]
Can't draw/repay any usdx amount under the collateral asset's debt limit. The debt limit should only be applied to creation and liquidation.

```    "logs": [
        {
            "msg_index": 0,
            "success": false,
            "log": "{\"codespace\":\"cdp\",\"code\":15,\"message\":\"proposed cdp debt of 9598560usdx is below the minimum of 10000000\"}",
            "events": [
                {
                    "type": "message",
                    "attributes": [
                        {
                            "key": "action",
                            "value": "repay_cdp"
                        }
                    ]
                }
            ]
        }
    ],```

**Comment by karzak** [comment]
> Can't draw/repay any usdx amount under the collateral asset's debt limit. The debt limit should only be applied to creation and liquidation.

Can you clarify by what you mean by 'the debt limit should only be applied to liquidation'? 

As far as the behavior, this is intended. The debt limit is enforced for repayment to protect against the following scenario:
1. Open a cdp for minimum debt
2. Repay debt below the debt floor, to some level which either 1) allows you to get around fees or 2) allows you to grind auctions because it's not economical for anyone to bid on them.

These theoretical attacks need to be balanced against the notably worse user experience from not being able to make otherwise valid repayments, but I'm okay starting with this configuration.

File: x/cdp/client/rest/query.go (hunk 1 above)
Comment (karzak) [reply x/cdp/client/rest/query.go]: The current query API basically separates queries that return one or more CDP object to the `cdps` prefix.  Queries that return information about the cdp module are at `cdp`. Do you want to do 

```
	r.HandleFunc("/cdp/params", getParamsHandlerFn(cliCtx)).Methods("GET")
	r.HandleFunc(fmt.Sprintf("/cdp/cdps/cdp/{%s}/{%s}", types.RestOwner, types.RestCollateralDenom), queryCdpHandlerFn(cliCtx)).Methods("GET")
	r.HandleFunc(fmt.Sprintf("/cdp/cdps/denom/{%s}", types.RestCollateralDenom), queryCdpsHandlerFn(cliCtx)).Methods("GET")
	r.HandleFunc(fmt.Sprintf("/cdp/cdps/ratio/{%s}/{%s}", types.RestCollateralDenom, types.RestRatio), queryCdpsByRatioHandlerFn(cliCtx)).Methods("GET")
	r.HandleFunc(fmt.Sprintf("/cdp/cdps/cdp/deposits/{%s}/{%s}", types.RestOwner, types.RestCollateralDenom), queryCdpDepositsHandlerFn(cliCtx)).Methods("GET")
```

It's verbose, but it seems logical and follows the pattern of prefixing everything with the module name used by `cosmos-sdk` modules. Would you like to see something else?

Comment (rhuairahrighairidh) [reply x/cdp/client/rest/query.go]: I would vote for your suggestion only just to align with the sdk. I agree that it's verbose and kind of weird doing `cdp/cdps/...`.

File: x/cdp/client/rest/query.go [hunk 2]
```diff
@@ -16,11 +16,11 @@ import (
...
-	r.HandleFunc(fmt.Sprintf("/cdp/{%s}/{%s}", types.RestOwner, types.RestCollateralDenom), queryCdpHandlerFn(cliCtx)).Methods("GET")
-	r.HandleFunc(fmt.Sprintf("/cdp/{%s}", types.RestCollateralDenom), queryCdpsHandlerFn(cliCtx)).Methods("GET")
-	r.HandleFunc(fmt.Sprintf("/cdp/{%s}/ratio/{%s}", types.RestCollateralDenom, types.RestRatio), queryCdpsByRatioHandlerFn(cliCtx)).Methods("GET")
-	r.HandleFunc(fmt.Sprintf("/cdp/deposits/{%s}/{%s}", types.RestOwner, types.RestCollateralDenom), queryCdpDepositsHandlerFn(cliCtx)).Methods("GET")
 	r.HandleFunc("/cdp/params", getParamsHandlerFn(cliCtx)).Methods("GET")
+	r.HandleFunc(fmt.Sprintf("/cdp/cdps/cdp/{%s}/{%s}", types.RestOwner, types.RestCollateralDenom), queryCdpHandlerFn(cliCtx)).Methods("GET")
+	r.HandleFunc(fmt.Sprintf("/cdp/cdps/denom/{%s}", types.RestCollateralDenom), queryCdpsHandlerFn(cliCtx)).Methods("GET")
+	r.HandleFunc(fmt.Sprintf("/cdp/cdps/ratio/{%s}/{%s}", types.RestCollateralDenom, types.RestRatio), queryCdpsByRatioHandlerFn(cliCtx)).Methods("GET")
+	r.HandleFunc(fmt.Sprintf("/cdp/cdps/cdp/deposits/{%s}/{%s}", types.RestOwner, types.RestCollateralDenom), queryCdpDepositsHandlerFn(cliCtx)).Methods("GET")
```
[the remark this answers is not in the document]
Comment (karzak) [reply x/cdp/client/rest/query.go]: The reason to have the additional keyword (cdp, denom, ratio), is that rmux can't differentiate between 
```
r.HandleFunc(fmt.Sprintf("/cdp/cdps/{%s}/{%s}", types.RestOwner, types.RestCollateralDenom...
r.HandleFunc(fmt.Sprintf("/cdp/cdps/{%s}", types.RestCollateralDenom...
```

as far as I can tell. If there's a way to simplify, happy to do so. I'll leave it to a follow on PR for now.

**Pull request merged**